core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -75,6 +75,13 @@ private[spark] class CoarseMesosSchedulerBackend(
   val coresByTaskId = new HashMap[String, Int]
   var totalCoresAcquired = 0
 
+  // Ports acquired so far
+  // SlaveID ->  ports
+  val takenPortsPerSlave = new mutable.HashMap[String, List[Long]]
```

Comment:
For now it is used by checkPorts utility method only, to check for the case where spark.executor.port or spark.blockManager.port have been defined and reserved on a slave and if so this port limits the number of execturs as we have discussed . So i keep track of what is allocated so i can deny an offer fast. Of course i could just check if those values are contained in the offered ports anyway. I guess i could refactor this part and ommit this state keeping if i dont miss anything here.